promotions/lib/controllers/backend/solidus_promotions/admin/promotions_controller.rb

```diff
@@ -11,8 +11,7 @@ class PromotionsController < BaseController
 
       def create
         @promotion = model_class.new(permitted_resource_params)
-        @promotion.codes.new(value: params[:single_code]) if params[:single_code].present?
-
+        @promotion.codes.new(promotion: @promotion, value: params[:single_code]) if params[:single_code].present?
```

Comment:
I think we can solve this by setting the `inverse_of` instead on the association. (I opened a PR that does just that when I encountered this error locally: https://github.com/solidusio/solidus/pull/5953)

Reply:
Sweet, thank you!